bitcoin/src/amount.rs

```diff
@@ -517,7 +517,7 @@ impl Amount {
     pub const fn from_sat(satoshi: u64) -> Amount { Amount(satoshi) }
 
     /// Gets the number of satoshis in this [`Amount`].
-    pub fn to_sat(self) -> u64 { self.0 }
+    pub const fn to_sat(self) -> u64 { self.0 }
```

Comment:
This change is in commit: `8e574a8e fee_rate: Make inner field explicit and use Amount` but is totally unrelated as far as I can tell. Can you pull it out into a separate patch so we can debate the pros/cons of making functions `const`? AFAIK we have no consensus on what functions should and should not be `const` yet.

Reply:
Purely based on the purpose of `const`, every function that we're sure will never perform FFI calls or need to do advanced optimizations like inline assembly/simd/... should have `const`. This one definitely applies. Agree with not mixing different things in a single commit.

Reply:
> every function that we're sure will never perform FFI calls or need to do advanced optimizations like inline assembly/simd/

That is a pretty wide net to cast, what about the additional complexity of not using non-const parts of the language (eg, `map`)?

Reply:
That complexity sucks but I expect it to be temporary as Rust improves. That being said, constructors should be top priority and const even at that cost since they tend to be used in `static`/`const`.

Reply:
That's a good point, I can definitely get behind "const all constructors".